src/planner/constify_now.c

```diff
@@ -0,0 +1,202 @@
+/*
+ * This file and its contents are licensed under the Apache License 2.0.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-APACHE for a copy of the license.
+ */
+
+#include <postgres.h>
+#include <access/xact.h>
+#include <datatype/timestamp.h>
+#include <nodes/makefuncs.h>
+#include <optimizer/optimizer.h>
+#include <utils/fmgroids.h>
+
+#include "cache.h"
+#include "dimension.h"
+#include "hypertable.h"
+#include "hypertable_cache.h"
+#include "planner.h"
+
+/*
+ * This implements an optimization to allow now() expression to be
+ * used during plan time chunk exclusions. Since now() is stable it
+ * would not normally be considered for plan time chunk exclusion.
```

Comment:
Is it currently used for runtime chunk exclusion? But for some reason it's much less effective than plan time chunk exclusion, so that's why this PR gives huge speedup?

Reply:
correct